aten/src/ATen/native/UpSample.h

```diff
@@ -0,0 +1,249 @@
+#include <math.h>
+
+#include <ATen/ATen.h>
+
+namespace at {
+namespace native {
+
+static inline void check_dim_size(
+    const Tensor& data,
+    int64_t dim,
+    int64_t dim_size,
+    int64_t size) {
+  /* Check dimension size of a tensor */
+  AT_CHECK(
+      data.dim() == dim && data.size(dim_size) == size,
+      "Expected tensor of dimension ",
+      dim,
+      " and tensor.size[",
+      dim_size,
+      "] == ",
+      size,
+      " but got: dimension ",
+      data.dim(),
+      " and tensor.size[",
+      dim_size,
+      "] = ",
+      data.size(dim_size));
+}
```

Comment:
Hmm, we should probably make some official replacement for `THNN_CHECK_DIM_SIZE` and put in the wiki. Doesn't have to be this PR.

Reply:
sounds good!